packages/shared/ReactTypes.js

```diff
@@ -81,54 +81,10 @@ export type RefObject = {|
   current: any,
 |};
 
-export type ReactEventResponderEventType =
-  | string
-  | {name: string, passive?: boolean};
-
-export type ReactEventResponder = {
-  targetEventTypes?: Array<ReactEventResponderEventType>,
-  rootEventTypes?: Array<ReactEventResponderEventType>,
-  createInitialState?: (props: null | Object) => Object,
-  allowMultipleHostChildren: boolean,
-  onEvent?: (
-    event: ReactResponderEvent,
-    context: ReactResponderContext,
-    props: null | Object,
-    state: null | Object,
-  ) => void,
-  onEventCapture?: (
-    event: ReactResponderEvent,
-    context: ReactResponderContext,
-    props: null | Object,
-    state: null | Object,
-  ) => void,
-  onRootEvent?: (
-    event: ReactResponderEvent,
-    context: ReactResponderContext,
-    props: null | Object,
-    state: null | Object,
-  ) => void,
-  onMount?: (
-    context: ReactResponderContext,
-    props: null | Object,
-    state: null | Object,
-  ) => void,
-  onUnmount?: (
-    context: ReactResponderContext,
-    props: null | Object,
-    state: null | Object,
-  ) => void,
-  onOwnershipChange?: (
-    context: ReactResponderContext,
-    props: null | Object,
-    state: null | Object,
-  ) => void,
-};
-
 export type ReactEventComponentInstance = {|
   currentFiber: mixed,
   props: null | Object,
-  responder: ReactEventResponder,
+  responder: mixed,
```

Comment:
I believe you should be able to reference a type defined in a host config here. Similar to how we import Instance in the reconciler even though the actually type is renderer specific.

https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberCommitWork.js#L11
https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMHostConfig.js#L85

So you could potentially reexport it through the host config to get to it from here.

That way `yarn flow dom` tests this more strictly. Other host configs would have to export something else here. Like any or mixed.

Reply:
Awesome, thank you! This helps a lot. I do wonder about the react-native sync though. I think they copy over the ReactTypes.js as is, which would now break. @TheSavior maybe we can change how this process is done slightly (you pointed this out to me last time), as now this file depends on `react` repo dependencies. Maybe when doing the RN sync, we can change these fields to any, as these are all build artifacts anyway?